### DISCUSSION TITLE
How to update kernel and dtb on jetson orin devkit?

### DISCUSSION THREAD

**Discussion by om26er** [OE4T/meta-tegra#1771] [Q&A]
For OTA of the rootfs we essentially write the squashfs image on the non-active partition, call nvbootctrl and reboot.

What is the "recommended" way to update the kernel/dtb ? I tried to `dd` kernel to the relevant partition but ended up breaking boot. Do I need to run a special command to convert the `Image-p3737-0000-p3701-0005.bin` image?

Any pointers in the correct direction would be helpful

**Comment by dwalkes** [comment]
There's an example at https://github.com/OE4T/tegra-demo-distro/blob/master/layers/meta-tegrademo/dynamic-layers/meta-swupdate/recipes-demo/images/swupdate-image-tegra/sw-description which maps yocto variables to partitions for the swupdate demo which might be helpful.

See also the discussion at https://github.com/OE4T/meta-tegra/discussions/1670#discussioncomment-10423161 about whether the partitions are used or just backup for dtb.
